src/search/search.component.ts

```diff
@@ -0,0 +1,220 @@
+import {
+	Component,
+	Input,
+	EventEmitter,
+	Output
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+/**
+ * Used to emit changes performed on search components.
+ * @export
+ * @class SearchChange
+ */
+export class SearchChange {
+	/**
+	 * Contains the `Search` that has been changed.
+	 * @type {Search}
+	 * @memberof SearchChange
+	 */
+	source: Search;
+	/**
+	 * The value of the `Search` field encompassed in the `SearchChange` class.
+	 * @type {string}
+	 * @memberof SearchChange
+	 */
+	value: string;
+}
+
+/**
+ * @export
+ * @class Search
+ * @implements {ControlValueAccessor}
+ */
+@Component({
+	selector: "ibm-search",
+	template: `
+		<div class="bx--form-item">
+			<div
+				class="bx--search"
+				[ngClass]="{
+					'bx--search--sm': size === 'sm',
+					'bx--search--lg': size === 'lg',
+					'bx--search--light': theme === 'light'
+				}"
+				role="search">
+				<label class="bx--label" [for]="id">Search</label>
+				<input
+					class="bx--search-input"
+					type="text"
+					role="search"
+					[id]="id"
+					[value]="value"
+					[placeholder]="placeholder"
+					[disabled]="disabled"
+					[required]="required"
+					[attr.aria-labelledby]="ariaLabelledby"
```

Comment:
Isn't this input labeled by the "Search" label?
If so, label should get an id and we don't need `ariaLabelledby`, it doesn't have to be an input. If provided, it clashes with label's `for` anyway.